packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -186,57 +217,99 @@ class Border {
     assert(bottom != null);
     assert(left != null);
 
-    Paint paint = new Paint();
+    Paint paint = new Paint()
+      ..strokeWidth = 0.0; // used for hairline borders
```

Comment:
It's so that I don't have to set it whenever I set the style to stroke. I can move it to the four branches below if that'd be better. (I believe it's ignored in fill mode, which we use if not doing a hairline.)